test/controllers/passengers_controller_test.rb

```diff
@@ -1,31 +1,175 @@
+      get passenger_path(passenger.id)
+
+      must_respond_with :success
+    end
+
+    it "responds with not found with an invalid driver id" do  #Passing
+
+    get passenger_path(-1)
+
+      must_respond_with :not_found
+    end
   end
 
   describe "new" do
-    # Your tests go here
+    it "responds with success" do
+      # See create
+    end
   end
 
   describe "create" do
-    # Your tests go here
+    it "can create a new driver with valid information accurately, and redirect" do #PASSING
+    new_passenger_hash = {
+        passenger:{
+            name: "Lip Gloss Poppin",
+            phone_num: "call ( me ) beep me"
+        }
+    }
+
+    expect {
+      post passengers_path, params: new_passenger_hash
+    }.must_change "Passenger.count", 1
+
+    new_passenger = Passenger.find_by(name: new_passenger_hash[:passenger][:name])
+    expect(new_passenger.phone_num).must_equal new_passenger_hash[:passenger][:phone_num]
+
+
+    must_respond_with :redirect
+    must_redirect_to passenger_path(new_passenger.id)
+    end
+
+    it "does not create a driver if the form data violates Driver validations, and responds with a redirect" do  #PASSING
+    driver_hash = {
+        driver: {
+            name: nil,
+            vin: nil,
+        },
+    }
+                                                                                                                 # Act-Assert
+    expect {
+      post drivers_path, params: driver_hash
+    }.wont_change "Driver.count"
+                                                                                                                 # Assert
+    must_respond_with :not_found
+    end
   end
 
   describe "edit" do
-    # Your tests go here
+    it "responds with success when getting the edit page for an existing, valid passenger" do
+      get edit_passenger_path(passenger.id)
+
+      must_respond_with :success
+    end
+
+    it "responds with redirect when getting the edit page for a non-existing passenger" do
+      get edit_passenger_path(-10)
+
+      # Act
+      must_respond_with :redirect
+      must_redirect_to passengers_path
+    end
   end
 
   describe "update" do
-    # Your tests go here
-  end
+    it "can update an existing passenger with valid information accurately, and redirect" do
+      skip
+      old_hash = passenger
+      updated_hash = {
+          passenger: {
+              name: "Lip Gloss Poppin",
+              phone_num: "call ( me ) beep me"
+          }
+      }
+
+      expect {
+        patch passenger_path(@passenger.id), params: updated_hash
+      }.wont_change "Passenger.count"
+
+      updated_passenger = Passenger.find_by(id: old_hash.id)
+      expect(old_hash.name).must_equal updated_passenger[:passenger][:name]
+      expect(old_hash.phone_num).must_equal updated_passenger[:passenger][:phone_num]
+
+      must_respond_with :redirect
+      must_redirect_to passenger_path(old_hash.id)
```

Comment:
This would fix this:

```suggestion
      old_hash = passenger
      updated_hash = {
          passenger: {
              name: "Lip Gloss Poppin",
              phone_num: "call ( me ) beep me"
          }
      }

      expect {
        patch passenger_path(old_hash.id), params: updated_hash
      }.wont_change "Passenger.count"

      updated_passenger = Passenger.find_by(id: old_hash.id)
      expect(updated_passenger.name).must_equal updated_hash[:passenger][:name]
      expect(updated_passenger.phone_num).must_equal updated_hash[:passenger][:phone_num]

      must_respond_with :redirect
      must_redirect_to passenger_path(old_hash.id)
```